src/VolumeRenderImpl.ts

```diff
@@ -1,38 +1,24 @@
-import { Euler, Object3D, Vector3 } from "three";
+import { Object3D } from "three";
 import { ThreeJsPanel } from "./ThreeJsPanel";
+import { SettingsFlags, VolumeRenderSettings } from "./VolumeRenderSettings";
+import { FuseChannel } from "./types";
+import Channel from "./Channel";
 
 export interface VolumeRenderImpl {
+  /**
+   * Applies the given VolumeRenderSettings to this volume renderer.
+   * @param settings a VolumeRenderSettings object to update values from.
+   * @param _dirtyFlags bitwise flag used to mark groups of changed settings in the
+   * provided `settings` object.
+   * If unset, forces recompute of all settings-based renderer configuration.
+   * See the `SettingsFlags` enum for recognized values.
+   */
+  updateSettings: (settings: VolumeRenderSettings, _dirtyFlags?: number | SettingsFlags) => void;
+
   get3dObject: () => Object3D;
-  setRayStepSizes: (_rayStepSize: number, _secondaryRayStepSize: number) => void;
-  setScale: (_scale: Vector3) => void;
-  setOrthoScale: (_scale: number) => void;
-  setResolution: (_x: number, _y: number) => void;
-  setAxisClip: (_axis: "x" | "y" | "z", _minval: number, _maxval: number, _isOrthoAxis: boolean) => void;
-  setIsOrtho: (_isOrtho: boolean) => void;
-  setOrthoThickness: (_thickness: number) => void;
-  setInterpolationEnabled: (_enabled: boolean) => void;
-  setGamma: (_gmin: number, _glevel: number, _gmax: number) => void;
-  setFlipAxes: (_flipX: number, _flipY: number, _flipZ: number) => void;
   doRender: (_canvas: ThreeJsPanel) => void;
   cleanup: () => void;
-  onChannelData: (_batch: number[]) => void;
```

Comment:
just noticing this.  where did `onChannelData` end up?

Reply:
None of the VolumeRenderImpl's implemented this method, so I'd removed it as part of the interface cleanup. Do you think it's worth keeping?

Reply:
if no one was using it then it can certainly go away :)